pom.xml

```diff
@@ -65,7 +65,8 @@
   </scm>
 
   <properties>
-    <jenkins.version>2.16</jenkins.version>
+    <jenkins.version>2.54</jenkins.version>
+    <java.level>8</java.level>
```

Comment:
If the new version of bcpkix-jdk15on doesn't require Java 8, I'd prefer to leave the jenkins baseline and Java version as-is. Yes, there are only ~100 instances using the latest version of this plugin on Jenkins 2.53 or older based on the [usage stats](http://stats.jenkins.io/pluginversions/bouncycastle-api.html), so it probably doesn't matter, but we don't really gain anything by updating the baseline so we might as well leave it. (I'd feel differently if upgrading allowed us to simplify code or make other improvements.)

The parent pom update is fine though, I think you'd just need to specify `<java.level>7</java.level>` explicitly.

Reply:
Yeah, but I feel it's time to stop thinking it's meaningful to keep supporting Java 7. I don't have 7 installed on my machine to build with it. It's been over three years since Oracle stopped supporting 7 and it doesn't receive any security fixes. 

I'm not committed to moving it forward to Java 8. If people think it is important to stay with 7, I can be okay with that, but if there aren't good reasons I think we'd be better to start moving forward.

Reply:
My point is just that it isn't like the current Jenkins or Java versions are slowing down development of this plugin. If moving the Jenkins or Java versions gave us any reliability improvements or new features through the use of new APIs, or if we had weird Java 7-specific hacks, or if users were reporting bugs specific to Java 7, then I would totally agree we should update, but otherwise it seems fine to just leave things as-is in the off chance that a user is paying for private Java 7 security updates and Jenkins 2.60.x or older security fixes.

Overall I'd say I am -0.5 on updating just because I don't see any motivation to do so, but if you or someone else feels strongly otherwise then I am fine with updating (although I'd go straight to the 2.60.3 Jenkins LTS version).

Reply:
I think this is probably the most recurring discussion in the community: to stay as-is or to move forward. I'm tempted to agree with Jeff (he has perfectly put into words my thoughts), but I have no strong opinion. Let's open the debate. For the moment: +1 on updating.

Reply:
I'm not as concerned with moving the Jenkins version. I'm open to whatever recommendations people have on that one.

It's possible that some company has purchased extended support for Java 7. I've been on projects before where we found ourselves in that situation. We always tried to extricate ourselves from that quickly. And we recognized that the world would be moving on without us and we would have to be very careful when updating to new versions of libraries. I think Oracle drove the stake hard into the ground over three years ago and we don't need to make any extra effort to coddle companies along from there. If they've purchased extended 7 support they've committed themselves to all the restrictions involved in that.

I'd say I'm at about a +1 on moving Java major version forward, but I'm interested in the discussion and opinions.

Reply:
I have no problems with the PR other than preferring to use the 2.60.3 LTS instead of 2.54, my point was just that updating the versions doesn't seem like it changes anything for this plugin.